dnSpy/dnSpy/Documents/Tabs/DefaultDocumentList.cs

```diff
@@ -48,6 +49,60 @@ sealed class DefaultDocumentList {
 		public void Add(DsDocumentInfo file) => files.Add(file);
 	}
 
+	sealed class DotnetReferenceFileFinder {
+		readonly CancellationToken cancellationToken;
+		readonly List<DefaultDocumentList> allLists;
+
+		public DotnetReferenceFileFinder(CancellationToken cancellationToken) {
+			this.cancellationToken = cancellationToken;
+			allLists = new List<DefaultDocumentList>();
+		}
+
+		public IEnumerable<DefaultDocumentList> AllFiles => allLists.ToArray();
+
+		public void Find() {
+			cancellationToken.ThrowIfCancellationRequested();
+
+			if (!TryGetRuntimeFolder(out var runtimesFolder)) {
+				return;
+			}
+
+			foreach (var appDir in Directory.EnumerateDirectories(runtimesFolder)) {
+				foreach (var versionDir in Directory.EnumerateDirectories(appDir)) {
+					var appName = $"{Path.GetFileName(appDir)} {Path.GetFileName(versionDir)}";
```

Comment:
A call to `cancellationToken.ThrowIfCancellationRequested();` inside these loops would allow earlier cancelation of the search effort when it is requested.